src/App.jsx

```diff
@@ -14,33 +16,83 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [sortedState, setSortedState] = useState(0);
+  const [reverseList, setReverseList] = useState(false);
+  const [changedGoods, setChangedGoods] = useState(goodsFromServer);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const handleAlphaSort = () => {
+    if (sortedState !== 1) {
+      setSortedState(1);
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+      if (reverseList) {
+        setChangedGoods(
+          [...changedGoods].sort((a, b) => a.localeCompare(b)).reverse(),
+        );
+      } else {
+        setChangedGoods([...changedGoods].sort((a, b) => a.localeCompare(b)));
+      }
+    }
```

Comment:
`[...changedGoods].sort((a, b) => a.localeCompare(b))` is performed unconditionally; why not process it, store it in a variable, and then apply `reverse` only if `reverseList is true`